pkg/oci/spec_opts.go

```diff
@@ -1644,30 +1643,13 @@ func WithWindowsNetworkNamespace(ns string) SpecOpts {
 	}
 }
 
-// WithCDIDevices injects the requested CDI devices into the OCI specification.
+// WithCDIDevices should be used from the cdi package. This version is used for
+// compatibility to point to the non-deprecated version but will return an error if used.
+// This function will be removed in 2.1.
+//
+// Deprecated: Use [github.com/containerd/containerd/v2/pkg/cdi.WithCDIDevices] instead.
 func WithCDIDevices(devices ...string) SpecOpts {
 	return func(ctx context.Context, _ Client, c *containers.Container, s *Spec) error {
-		if len(devices) == 0 {
-			return nil
-		}
-
-		if err := cdi.Refresh(); err != nil {
-			// We don't consider registry refresh failure a fatal error.
-			// For instance, a dynamically generated invalid CDI Spec file for
-			// any particular vendor shouldn't prevent injection of devices of
-			// different vendors. CDI itself knows better and it will fail the
-			// injection if necessary.
-			log.G(ctx).Warnf("CDI registry refresh failed: %v", err)
-		}
-
-		if _, err := cdi.InjectDevices(s, devices...); err != nil {
-			return fmt.Errorf("CDI device injection failed: %w", err)
-		}
-
-		// One crucial thing to keep in mind is that CDI device injection
-		// might add OCI Spec environment variables, hooks, and mounts as
-		// well. Therefore it is important that none of the corresponding
-		// OCI Spec fields are reset up in the call stack once we return.
-		return nil
+		return errors.New("must use cdi package for CDI device injection")
```

Comment:
The function should remain functional, as it is only "deprecated"?

Reply:
Or maybe just remove the function, if the impact is considered low.

Reply:
I'm ok removing completely